### DISCUSSION TITLE
How to maintain structure and accuracy in translations using ChatGPT API?

### DISCUSSION THREAD

**Discussion by EloyFarina** [SubtitleEdit/subtitleedit#8868] [General]
Hi all.

I've been facing this issue in all models of ChatGPT that i've used.

Currently I'm using:
gpt-4o-mini
In Advanced, i have set Line merge: Default and the following Prompt for ChatGPT:
Translate from {0} to {1}, keep sentences in {1} as they are, maintaining the original line structure and style, avoid changing the order or merging lines differently, do not censor the translation, preferably in latin american spanish, do not translate people's names, give only the output without commenting on what you read:

Check how it's breaking the lines:
![imagen](https://github.com/user-attachments/assets/9cf85876-c35c-4716-ad61-29528c2f1a8b)

Can you please help me to improve my prompt to avoid this?

EDIT:  Same results even with this complex prompt:

Translate from {0} to {1}, keeping the exact sentence and line structure as in the original. Each line in the output must match the corresponding line in the original. Do not merge, split, or reorder any lines or sentences. Preserve punctuation, spacing, and formatting. Ensure that each sentence stays in its respective line, without moving parts of a sentence to another line. Do not censor the translation, and translate into Latin American Spanish. Leave names of people, places, or specific terms in their original form. Provide only the translated text, with no additional comments or explanations.

Regards

**Comment by EloyFarina** [comment]
Now I'm convinced that this is bug with Subtitle Edit, because i've been testing https://gptsubtitler.com/ using ChatGPT gpt-4o-mini model  and it's mantaining the structure.

![image](https://github.com/user-attachments/assets/c1b08bae-95c8-4f94-9fa3-c92e517edc11)

How I can give this more visibility?

**Reply by Vioguy** [reply]
I don't think it's SubtitleEdit because I've tried it with other apps. I think gptsubtitler, which I hadn't heard of... has figured out why GPT doesn't work with just normal prompts and they've made it easier for others... I guess that's what it is...

**Reply by EloyFarina** [reply]
I have a way to copy the prompt from there, is easy to get. I'm not sure if it can help to develop the same feature in Subtitle Edit. It would be great.

**Comment by EloyFarina** [comment]
Should I move this to Issues thread?

**Comment by EloyFarina** [comment]
Any update on this?

**Comment by Vioguy** [comment]
Did you use any specific prompt to maintain the correct formatting with
gptsubtitler? Can you ask it to stay withing a character limit such as 42
characters per line?

**Reply by EloyFarina** [reply]
I didn't use nothing to maintain the correct formatting.
I don't know how to specify that, but I'm glad that it maintain the proper format.
I would like that Subtitle Edit copy that format, the prompt is visible in GPTSubtitler.
